cmd/appliance/upgrade/cancel.go

```diff
@@ -139,65 +144,56 @@ func upgradeCancelRun(cmd *cobra.Command, args []string, opts *upgradeCancelOpti
 				appliancepkg.UpgradeStatusReady,
 				appliancepkg.UpgradeStatusFailed,
 			}
-			// wg is the wait group for the progressbars
-			wg           sync.WaitGroup
-			errorChannel = make(chan error, count)
 		)
-		cancelProgressBars := mpb.New(mpb.WithOutput(spinnerOut), mpb.WithWaitGroup(&wg))
-		wg.Add(count)
 		retryCancel := func(ctx context.Context, appliance openapi.Appliance) error {
 			return backoff.Retry(func() error {
 				return a.UpgradeCancel(ctx, appliance.GetId())
 			}, backoff.NewExponentialBackOff())
 		}
 
-		for _, ap := range appliances {
-			appliance := ap
-			qw.Push(appliance)
-			statusReport := make(chan string)
-			go a.UpgradeStatusWorker.Watch(ctx, cancelProgressBars, appliance, appliancepkg.UpgradeStatusIdle, appliancepkg.UpgradeStatusReady, statusReport)
+		type queueStruct struct {
+			appliance openapi.Appliance
+			status    chan<- string
+		}
 
-			go func(appliance openapi.Appliance) {
-				defer func() {
-					wg.Done()
-					close(statusReport)
-				}()
-				if err := a.UpgradeStatusWorker.Subscribe(ctx, appliance, wantedStatus, undesiredStatus, statusReport); err != nil {
-					errorChannel <- err
-				}
-			}(appliance)
+		var bars *tui.Progress
+		if !opts.ciMode {
+			bars = tui.NewProgress(ctx, spinnerOut)
+			defer bars.Wait(3 * time.Second)
+		}
+		for _, appliance := range appliances {
+			var s chan<- string
+			if !opts.ciMode {
+				var t *tui.Tracker
+				t, s = bars.AddTracker(appliance.GetName(), "cancelled")
```

Comment:
```suggestion
				t, s := bars.AddTracker(appliance.GetName(), "cancelled")
```

Reply:
will not work. the `s` chan variable need to passed down, either as nil or as a channel, so we cannot declare `s` as a new variable.